src/angularfire2.ts

```diff
@@ -23,64 +24,72 @@ export class AngularFire {
   list: (url:string, opts?:FirebaseListFactoryOpts) => FirebaseListObservable<any[]>;
   object: (url: string, opts?:FirebaseObjectFactoryOpts) => FirebaseObjectObservable<any>;
   constructor(
-    @Inject(FirebaseUrl) private fbUrl:string,
-    public auth:FirebaseAuth,
+    @Inject(FirebaseConfig) private fbUrl:string,
+    public auth:AngularFireAuth,
     public database: FirebaseDatabase) {}
-
 }
 
-function getAbsUrl (root:string, url:string) {
+function getAbsUrl (root:FirebaseAppConfig, url:string) {
   if (!(/^[a-z]+:\/\/.*/.test(url))) {
     // Provided url is relative.
-    url = root + url;
+    url = root.databaseURL + url;
   }
   return url;
 }
 
 export const COMMON_PROVIDERS: any[] = [
-  provide(FirebaseRef, {
-    useFactory: (url:string) => new Firebase(url),
-    deps: [FirebaseUrl]}),
-  FirebaseAuth,
+  provide(FirebaseApp, {
+    useFactory: (config: FirebaseAppConfig) => {
+      return initializeApp({
+        apiKey: config.apiKey,
+        authDomain: config.authDomain,
+        databaseURL: config.databaseURL,
+        storageBucket: config.storageBucket
+      });
+    },
+    deps: [FirebaseConfig]}),
+  AngularFireAuth,
   AngularFire,
   FirebaseDatabase
 ];
 
 export const FIREBASE_PROVIDERS:any[] = [
   COMMON_PROVIDERS,
   provide(AuthBackend, {
-    useFactory: (ref: Firebase) => new FirebaseSdkAuthBackend(ref, false),
-    deps: [FirebaseRef]
+    useFactory: (app: firebase.app.App) => new FirebaseSdkAuthBackend(app, false),
+    deps: [FirebaseApp]
   })
 ];
 
 /**
  * Used to define the default Firebase root location to be
  * used throughout an application.
  */
-export const defaultFirebase = (url: string): Provider => {
-  return provide(FirebaseUrl, {
-    useValue: url
+export const defaultFirebase = (config: FirebaseAppConfig): Provider => {
+  // remove a trailing slash from the Database URL if it exists
+  config.databaseURL = utils.stripTrailingSlash(config.databaseURL);
+  return provide(FirebaseConfig, {
+    useValue: config
   });
 };
 
 export {
-  FirebaseAuth,
+  AngularFireAuth,
   FirebaseDatabase,
   FirebaseListObservable,
   FirebaseObjectObservable,
   FirebaseListFactory,
   FirebaseObjectFactory,
   firebaseAuthConfig,
-  FirebaseAuthState,
   AuthMethods,
   AuthProviders
 }
 
-export {FirebaseUrl, FirebaseRef, FirebaseAuthConfig} from './tokens';
+export { AuthToken } from './providers/auth_backend';
```

Comment:
Should get rid of AuthToken type and just use string...unless I remember some reason not to.